lib/models/wallet_info_change_notifier.dart

```diff
@@ -0,0 +1,103 @@
+      lightningHistory: List.empty());
+
+  List<PaymentHistoryItem> history = List.empty();
+
+  Amount pending() {
+    OffChain offChain = walletInfo.balance.offChain;
+    OnChain onChain = walletInfo.balance.onChain;
+    return Amount(onChain.untrustedPending + onChain.trustedPending + offChain.pendingClose);
+  }
+
+  Amount totalOnChain() {
+    OnChain onChain = walletInfo.balance.onChain;
+    return Amount(onChain.confirmed + pending().asSats);
+  }
+
+  Future<void> update(WalletInfo walletInfo) async {
+    this.walletInfo = walletInfo;
+    var lth = walletInfo.lightningHistory.map((e) {
+      var amount = Amount(e.sats);
+      PaymentType type;
+      switch (e.flow) {
+        case Flow.Inbound:
+          type = PaymentType.receive;
+          amount = Amount(e.sats);
+          break;
+        case Flow.Outbound:
+          type = PaymentType.send;
+          amount = Amount(-e.sats);
+          break;
+      }
+      PaymentStatus status;
+      switch (e.status) {
+        case HTLCStatus.Failed:
+        case HTLCStatus.Expired:
+          status = PaymentStatus.failed;
+          break;
+        case HTLCStatus.Succeeded:
+          status = PaymentStatus.finalized;
+          break;
+        case HTLCStatus.Pending:
+          status = PaymentStatus.pending;
+          break;
+      }
+      return PaymentHistoryItem(amount, type, status, e.createdTimestamp, e);
+    }).toList();
+
+    var bph = walletInfo.bitcoinHistory.map((bitcoinTxHistoryItem) {
+      var amount = bitcoinTxHistoryItem.sent != 0
+          ? Amount((bitcoinTxHistoryItem.sent -
+                  bitcoinTxHistoryItem.received -
+                  bitcoinTxHistoryItem.fee) *
+              -1)
+          : Amount(bitcoinTxHistoryItem.received);
+
+      var type =
+          bitcoinTxHistoryItem.sent != 0 ? PaymentType.sendOnChain : PaymentType.receiveOnChain;
+
+      var status =
+          bitcoinTxHistoryItem.isConfirmed ? PaymentStatus.finalized : PaymentStatus.pending;
+      return PaymentHistoryItem(
+          amount, type, status, bitcoinTxHistoryItem.timestamp, bitcoinTxHistoryItem);
+    }).toList();
+
+    final combinedList = [...bph, ...lth];
+    combinedList.sort((a, b) => b.timestamp.compareTo(a.timestamp));
+    history = combinedList;
+
+    FLog.trace(text: 'Successfully synced payment history');
+    super.notifyListeners();
+  }
+
+  Future<void> refreshWalletInfo() async {
+    try {
+      final walletInfo = await api.refreshWalletInfo();
+      await update(walletInfo);
+      FLog.trace(text: 'Successfully refreshed wallet info');
+    } catch (error) {
+      FLog.error(text: "Failed to get wallet info:" + error.toString());
+    }
+  }
+
+  List<PaymentHistoryItem> bitcoinHistory() {
+    return history.where((element) => element.type.isBitcoin()).toList();
+  }
+
+  List<PaymentHistoryItem> lightningHistory() {
+    return history.where((element) => element.type.isLightning()).toList();
+  }
+}
```

Comment:
are you not planning to expose total `history` ? (if this is the case, there's little point in creating a single list with them, only to split them up again...)

Reply:
I just move the code as is into a single change notifier, since this data was now coming from one model `WalletInfo`.